# Data Analytics - Summative Assessment One

## Business Case
As a part of a software development team for Censored, I have been asked to conduct data-driven analysis of games on the online gaming marketplace Steam.  The purpose of this analysis is to make informed data-driven decisions on what type of game the organisation should produce in order to maximise audience reach, engagement, and cost-effectiveness.  

By examining existing games on Steam, this project aims to identify patterns in the games' popularity, reception, player engagement, tags, genres, or combinations of genres.  The aim of these insights is to create a game with the maximum reach possible to provide the greatest return on investment.  


## Data acquisition 





### Import Python Modules 🐍

In [17]:
import pandas as pd # Pandas is a module for data manipulation, it makes use of dataframe objects to store and manipulate data.
import numpy as np # Numpy is a module for numerical computing, it provides support for arrays and matrices and mathematical functions.
import matplotlib.pyplot as plt # Matplotlib is a module for data visualization.
import seaborn as sns # Seaborn is a module for statistical data visualization, built on top of Matplotlib.
import plotly.express as px # Plotly Express is a module for making interactive visualisations.

### Loading dataset 📥

In [18]:
steam_games_df = pd.read_csv("steam_games.csv")

This creates a pandas dataframe object from the origional csv file.  The reason the dataframe object are so effective for data analysis is that they have built in methods and functions which enable the user to quickly and easily inspect and transform the data.

### Dataset source and description
This analysis uses the [Steam Games Dataset](https://www.kaggle.com/datasets/fronkongames/steam-games-dataset), a public open dataset from Kaggle.  The dataset provides comprehensive information on the games sold on Steam, including commercial, technical and user engagement data.

## Initial data inspection 🧭

### Preview the data

In [19]:
steam_games_df.head()

,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,Discount,DLC count,About the game,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
0,2539430,Black Dragon Mage Playtest,"Aug 1, 2023",0 - 0,0,0,0.00,0,0,NaN,...,0,0,0,NaN,NaN,NaN,NaN,NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
1,496350,Supipara - Chapter 1 Spring Has Come!,"Jul 29, 2016",0 - 20000,0,0,5.24,65,0,"Springtime, April: when the cherry trees come ...",...,0,8,0,minori,MangaGamer,"Single-player,Steam Trading Cards,Steam Cloud,...",Adventure,"Adventure,Visual Novel,Anime,Cute",https://shared.akamai.steamstatic.com/store_it...,NaN
2,1034400,Mystery Solitaire The Black Raven,"May 6, 2019",0 - 20000,0,0,4.99,0,0,"Immerse yourself in the most beloved, mystical...",...,0,0,0,Somer Games,8floor,"Single-player,Family Sharing",Casual,"Casual,Card Game,Solitaire,Puzzle,Hidden Objec...",https://shared.akamai.steamstatic.com/store_it...,NaN
3,3292190,버튜버 파라노이아 - Vtuber Paranoia,"Oct 31, 2024",0 - 20000,1,0,8.99,0,1,"synopsis 'Hello, I'm Hiyoro, a new YouTuber!' ...",...,0,0,0,유진게임즈,유진게임즈,"Single-player,Steam Achievements,Family Sharing","Casual,Indie,Simulation",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
4,3631080,Maze Quest VR,"Apr 24, 2025",0 - 20000,0,0,4.99,0,0,Its not just a Maze; its a Quest! Enter the ca...,...,0,0,0,Reality Expanded LLC,Reality Expanded LLC,"Single-player,VR Only,Steam Leaderboards,Famil...","Action,Early Access",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN


On initially looking at the dataframes head it apears that there is a missing column name.  Due to 'supported languages' being languages, 'about the game' being text and price being in the format x.xx it makes sense that the 'DiscountDLC Count' should be two columns.  This is a mistake present in the kaggle dataset.  To fix 'DiscountDLC Count' was split into 'Discount' and 'DLC Count' by a comma.  
  
Overall this isn't very usful as it only shows a few columns potentially hiding errors.

In [20]:
steam_games_df.shape

(122611, 40)

The shape of the dataframe is 40 columns and 122611 rows.  This means that there are 40 columns and 122611 rows.  40 columns are too many for this investigation, any extra colums will just confuse.  Furthermore there are way to many rows for the data to ever be manually checked by a human. 

### Check columns

In [23]:
steam_games_df.columns

Index(['AppID', 'Name', 'Release date', 'Estimated owners', 'Peak CCU',
       'Required age', 'Price', 'Discount', 'DLC count', 'About the game',
       'Supported languages', 'Full audio languages', 'Reviews',
       'Header image', 'Website', 'Support url', 'Support email', 'Windows',
       'Mac', 'Linux', 'Metacritic score', 'Metacritic url', 'User score',
       'Positive', 'Negative', 'Score rank', 'Achievements', 'Recommendations',
       'Notes', 'Average playtime forever', 'Average playtime two weeks',
       'Median playtime forever', 'Median playtime two weeks', 'Developers',
       'Publishers', 'Categories', 'Genres', 'Tags', 'Screenshots', 'Movies'],
      dtype='str')

Column names are mostly self explanatory with the exeption of a few such as 'Posative' and 'Negative' which need aditional context to make sense.

### Check datatypes

In [24]:
steam_games_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 122611 entries, 0 to 122610
Data columns (total 40 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   AppID                       122611 non-null  int64  
 1   Name                        122610 non-null  str    
 2   Release date                122611 non-null  str    
 3   Estimated owners            122611 non-null  str    
 4   Peak CCU                    122611 non-null  int64  
 5   Required age                122611 non-null  int64  
 6   Price                       122611 non-null  float64
 7   Discount                    122611 non-null  int64  
 8   DLC count                   122611 non-null  int64  
 9   About the game              114162 non-null  str    
 10  Supported languages         122611 non-null  str    
 11  Full audio languages        122611 non-null  str    
 12  Reviews                     12070 non-null   str    
 13  Header image             

Overall the dataframe has a mix of quantitative and qualitative data.  Data type is mostly correct except columns like 'release date' which should be a date.  There are also a few unusable columns due to lack of data such as 'Movies', 'Notes' or 'Score Rank'.

In [25]:
steam_games_df.describe()

,AppID,Peak CCU,Required age,Price,Discount,DLC count,Metacritic score,User score,Positive,Negative,Score rank,Achievements,Recommendations,Average playtime forever,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Movies
count,1.226110e+05,1.226110e+05,122611.000000,122611.000000,122611.000000,122611.000000,122611.000000,122611.000000,1.226110e+05,1.226110e+05,40.000000,122611.000000,1.226110e+05,1.226110e+05,122611.000000,1.226110e+05,122611.000000,0.0
mean,1.985386e+06,5.459332e+01,0.167611,4.765091,18.353663,0.545856,2.564941,0.024549,1.044986e+03,1.691974e+02,99.175000,18.087015,9.618250e+02,2.080232e+02,13.789268,1.735705e+02,14.722170,NaN
std,1.087595e+06,3.729452e+03,1.653591,12.531030,28.858970,14.516026,13.660559,1.394901,2.809173e+04,5.374645e+03,0.675107,141.493879,2.187880e+04,1.121768e+04,270.378053,1.120254e+04,294.509615,NaN
min,1.000000e+01,0.000000e+00,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000e+00,0.000000e+00,98.000000,0.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
25%,1.063175e+06,0.000000e+00,0.000000,0.550000,0.000000,0.000000,0.000000,0.000000,0.000000e+00,0.000000e+00,99.000000,0.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
50%,1.907380e+06,0.000000e+00,0.000000,2.240000,0.000000,0.000000,0.000000,0.000000,5.000000e+00,1.000000e+00,99.000000,2.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
75%,2.869560e+06,0.000000e+00,0.000000,5.240000,40.000000,0.000000,0.000000,0.000000,3.700000e+01,1.000000e+01,100.000000,19.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
max,4.264350e+06,1.013936e+06,21.000000,999.980000,100.000000,3703.000000,97.000000,100.000000,7.642084e+06,1.173003e+06,100.000000,9821.000000,4.830455e+06,3.429544e+06,20088.000000,3.429544e+06,20088.000000,NaN


.describe() shows sumary statistics for numerical colums.  
Quite a few of the third quartile are 0, this is due to a large number of 0 results biasing the data some such as 'DLC Count' are probuably legitimate however for 'User score' and 'Metacritic score' no score is probuably recorded as 0.  
Columns like 'Positive', 'Negative' and 'Recomondation' columns have a median of 5, 1 and 0 respectivly but means of 1 x 10E3, 2 x 10E2 and 2 x 10E2, this large difference between the median and mean sugests a few extremly popular games with high engagment.
The 'movies' column contains no legitimate data and should be removed.

In [27]:
steam_games_df.describe(include="str")

,Name,Release date,Estimated owners,About the game,Supported languages,Full audio languages,Reviews,Header image,Website,Support url,Support email,Metacritic url,Notes,Developers,Publishers,Categories,Genres,Tags,Screenshots
count,122610,122611,122611,114162,122611,122611,12070,122530,49676,54142,100348,4256,22458,114174,113702,113658,114198,83346,116593
unique,121454,5081,14,113556,19113,3710,11884,122420,39703,35393,60515,4160,18619,70812,62684,13291,2894,77179,116483
top,Shadow of the Tomb Raider: Definitive Edition,"Oct 23, 2025",0 - 20000,Help the beautiful girls to immerse themselves...,['English'],[],"“🏆 I loved! ❤️❤️❤️” Player “🏆 Cute kittens, ea...",https://shared.akamai.steamstatic.com/store_it...,https://www.facebook.com/8FloorGames/,https://www.facebook.com/8FloorGames,info@bigfishgames.com,https://www.metacritic.com/game/pc/shadow-of-t...,This Game may contain content not appropriate ...,EroticGamesClub,BFG Entertainment,"Single-player,Family Sharing","Casual,Indie","Adventure,Casual,Hidden Object",https://shared.akamai.steamstatic.com/store_it...
freq,20,190,75404,65,55314,72730,43,20,274,328,323,20,181,231,554,23383,6700,273,20


.describe(include="str") shows sumary statistics for string colums.  
The name column contains 121454 unique vaues out of 122610 records, 'Shadow of the Tomb Raider: Definitive Edition' apears 20 times sugesting duplicate records in the dataset.  
Estimated owners shows a range of owners therfore for analysis it may be usful to split into minimum and maximum columns.    
The catagorical data columns 'Catagories', 'Genres' and 'Tags' contain comma seperated strings which requere processing before used in analysis.

### Check missing

## Data cleaning 🛀🏾

### Remove Unnesesary Columns

### Remove Duplicates

### Handle Missing Values

### Convert Datatypes

### Verify cleaning 🛀🏾

## Exploratory data analysis

### Summary statistics

### Graphs

## Statistical analysis

## Graphical and numerical reporting

### Audience focussed charts

### Main Findings

### Recommendations